app/main.py

```
@@ -2,4 +2,78 @@ class Cargo:
+        if coords is None:
+            coords = [0, 0]
+        self.name = name
+        self.weight = weight
+        self.coords = coords
+
+    def go_forward(self, step: int = 1) -> list[int]:
+        self.coords[1] += step
+        return self.coords
+
+    def go_back(self, step: int = 1) -> list[int]:
+        self.coords[1] -= step
+        return self.coords
+
+    def go_right(self, step: int = 1) -> list[int]:
+        self.coords[0] += step
+        return self.coords
+
+    def go_left(self, step: int = 1) -> list[int]:
+        self.coords[0] -= step
+        return self.coords
+
+    def get_info(self) -> str:
+        return f"Robot: {self.name}, Weight: {self.weight}"
+
+
+class FlyingRobot(BaseRobot):
+    def __init__(
+            self, name: str, weight: int, coords: list[int] = None
+    ) -> None:
+        super().__init__(
+            name, weight, coords if coords is not None else [0, 0, 0]
+        )
+
+    def go_up(self, step: int = 1) -> list[int]:
+        self.coords[2] += step
+        return self.coords
+
+    def go_down(self, step: int = 1) -> list[int]:
+        self.coords[2] -= step
+        return self.coords
+
+
+class DeliveryDrone(FlyingRobot):
+    def __init__(
+        self, name: str, weight: int, max_load_weight: int,
+        coords: list[int] = None, current_load: Cargo = None
+    ) -> None:
+        super().__init__(name, weight, coords)
+        self.max_load_weight = max_load_weight
+        self.current_load = current_load
+
+    def hook_load(self, cargo: Cargo) -> str:
+        if self.current_load is not None:
+            return "Cannot hook load. There is already cargo hooked."
+        elif cargo.weight > self.max_load_weight:
+            return (
+                f"Cannot hook load. The cargo is too heavy: "
+                f"{cargo.weight} > {self.max_load_weight}."
+            )
+        self.current_load = cargo
+        return f"Cargo weighing {cargo.weight} hooked successfully."
```

The `hook_load` method currently returns a string to indicate the status of the operation. This is not a common practice in Python. It would be better to raise an exception in case of an error. This way, the calling code can decide what to do in case of an error (e.g., show a message to the user, try again, etc.).